src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -184,7 +184,27 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
 
   override def clone() = new ArrayDeque(array.clone, start, end)
 
-  override def clear() = if (nonEmpty) set(array = ArrayDeque.alloc(ArrayDeque.DefaultInitialSize), start = 0, end = 0)
+  /**
+    * Note: This does not actually resize the internal representation.
+    * See clearAndShrink if you want to also resize internally
+    */
+  override def clear() = {
+    nullify()
+    start = 0
+    end = 0
+  }
```

Comment:
I am not a fan of sizeHint request being made as implicit behaviour changing. I think the current state of the API is fine. By default, both clear and removeFirst/removeLast APIs do not shrink. The `remove(x, count)` API does shrink but only if we are removing more than half the elements - which is a reasonable choice.

If the user wants to manage memory herself, she can either call `.trimToSize` explicitly at the end OR call `.clearAndShrink` instead of `.clear` and call `.removeFirst(autoShrink=true)` (which will shrink if size is less than half of allocation).